test/common/index.js

```diff
@@ -690,12 +690,26 @@ Object.defineProperty(exports, 'hasIntl', {
 });
 
 // Useful for testing expected internal/error objects
+const kAcceptableErrorTypes = ['NodeError', 'AssertionError', 'RangeError'];
 exports.expectsError = function expectsError({code, type, message}) {
-  return function(error) {
+  const expectedTypeName = type && type.name;
+  return function expectsErrorCheck(error) {
     assert.strictEqual(error.code, code);
-    if (type !== undefined)
-      assert(error instanceof type,
-             `${error} is not the expected type ${type}`);
+    if (expectedTypeName) {
```

Comment:
For the purposes of our own test suite, I'd expect a `TypeError` assertion to only work with `TypeError`s.